endpoints/openrtb2/auction.go

```diff
@@ -1334,6 +1339,36 @@ func validateDevice(device *openrtb2.Device) error {
 	return nil
 }
 
+func validateOrFillChannel(requestPrebid *openrtb_ext.ExtRequestPrebid, requestExt *openrtb_ext.RequestExt, reqWrapper *openrtb_ext.RequestWrapper, isAmp bool) error {
+	if requestPrebid != nil {
+		if requestPrebid.Channel != nil {
```

Comment:
I think the conditionals here can be simplified to something like:
```
if requestPrebid == nil || requestExt.GetPrebid().Channel == nil {
	fillChannel(requestPrebid, requestExt, reqWrapper, isAmp)
} else if requestPrebid.Channel.Name == "" {
	return errors.New("ext.prebid.channel.name can't be empty")
}
```

And then you can just create and empty `requestPrebid` in the `fillChannel` function if it is `nil`

Reply:
Thanks this is great!